users/vieting/jobs/fairseq.py

```diff
@@ -109,38 +103,95 @@ def create_tsv_and_labels(self):
-
-                # determine whether to write to train or valid
-                if rand.random() >= self.valid_percent:
-                    tsv_out = train_tsv
-                    ltr_out = train_ltr
-                    wrd_out = train_wrd
-                else:
-                    tsv_out = valid_tsv
-                    ltr_out = valid_ltr
-                    wrd_out = valid_wrd
                 
                 # write audio path to tsv files
-                print(f"{rel_audio_path}\t{frames}", file=tsv_out)
+                print(f"{rel_audio_path}\t{frames}", file=dest_tsv)
 
                 # write transcription to transcription files
                 print(
                     " ".join(list(audio_trans.replace(" ", "|"))) + " |",
-                    file=ltr_out,
+                    file=dest_ltr,
                 )
-                print(audio_trans, file=wrd_out)
-        
-        # close all files
-        valid_tsv.close()
-        valid_ltr.close()
-        valid_wrd.close()
-        
-        train_tsv.close()
-        train_ltr.close()
-        train_wrd.close()
+                print(audio_trans, file=dest_wrd)
+
+        dest_tsv.close()
+        dest_ltr.close()
+        dest_wrd.close()
 
+    def get_common_dir(self):
+        """
+        Returns the common directory of all audios given in the corpora.
+        """
+        common_dir = None
+        # iterate over all corpora
+        for corpus_path in self.corpus_paths:
+            corpus_object = corpus.Corpus()
+            corpus_object.load(corpus_path.get())
+            for segment in corpus_object.segments():
+                audio_path = segment.recording.audio
+                assert os.path.exists(audio_path), f"Path {audio_path} does not exist."
+                if common_dir is None:
+                    common_dir = os.path.dirname(audio_path)
+                else:
+                    common_dir = os.path.commonpath([common_dir, os.path.dirname(audio_path)])
+        return common_dir
 
 
+class MergeLabeledFairseqDataJob(Job):
+    """
+    Merge multiple labeled fairseq data directories into one. Each directory must contain the following files:
+    - <name>.tsv
+    - <name>.ltr
+    - <name>.wrd
+
+    If create_letter_dict is set to True, the following file will be created:
+    - dict.ltr.txt
+    """
+    def __init__(
+        self,
+        labeled_data_paths: Union[List[tk.Path], tk.Path],
+        create_letter_dict: bool = True,
+    ):
+        if not isinstance(labeled_data_paths, list):
+            labeled_data_paths = [labeled_data_paths]
+        self.labeled_data_paths = labeled_data_paths
+        assert all([isinstance(path, tk.Path) for path in self.labeled_data_paths])
+
+        self.create_letter_dict = create_letter_dict
+
+        self.out_task_path = self.output_path("task", directory=True)
+        self.out_dict_ltr_path = self.output_path("task/dict.ltr.txt")
+
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
```

Comment:
```suggestion
        self.rqmt = {"time": 1, "mem": 8, "cpu": 1}
```
One hour should be much more than enough I guess.